playback/src/decoder/symphonia_decoder.rs

```diff
@@ -40,7 +40,11 @@ impl SymphoniaDecoder {
         };
         let mss = MediaSourceStream::new(Box::new(input), mss_opts);
 
-        let format_opts = Default::default();
+        let format_opts = FormatOptions {
+            enable_gapless: true,
```

Comment:
In [config.rs](https://github.com/librespot-org/librespot/blob/new-api/playback/src/config.rs#L127) we have a configuration option for gapless. All it does is stop the sink before loading a new track in [player.rs](https://github.com/librespot-org/librespot/blob/new-api/playback/src/player.rs#L1602).

The OCD in my itches to make these lines follow that configuration option, but that's currently not passed to `Decoder::new` and can you confirm that it won't really matter?

Reply:
Hi @roderickvd,

This option asks a `FormatReader` to provide "trim" information in each `Packet` so that a `Decoder` knows how to trim the output and remove any silent delay and padding samples. It's required to be done this way because a transform based codec like MP3 and Vorbis can only produce fixed size blocks of samples. Thus, extra timing information is stored by the container to aid in trimming the output.

I set it to always be enabled because that preserves the existing behaviour of librespot which was to trim the decoded output (Lewton did this by default).

This is one required part of many for proper gapless playback. Keeping the sink open is another, but you would also need to be able to dynamically insert a resampler if dealing with tracks with differing sample rates. Also, the next song would need to be buffered and have samples decoded to truly allow a gapless transition between tracks.

Reply:
> I set it to always be enabled because that preserves the existing behaviour of librespot which was to trim the decoded output (Lewton did this by default).

Certainly that reaffirms that it's OK for the decoder to trim data regardless of `librespot`'s "global" gapless toggle.
 
> This is one required part of many for proper gapless playback. Keeping the sink open is another, but you would also need to be able to dynamically insert a resampler if dealing with tracks with differing sample rates. Also, the next song would need to be buffered and have samples decoded to truly allow a gapless transition between tracks.

We don't have to deal with different sample rates fortunately as Spotify is all 44.1 kHz, and I don't see that changing anytime soon. Buffering is already in by means of the preloading mechanism. Although in `player.rs` it takes a few CPU cycles to swap the loaded (finished) track with the preloaded track, the sink won't notice because it should have a big enough buffer to not completely drain in the meantime.

Thanks again and let's get this in!